### PR TITLE
Add Hamming exercise

### PR DESCRIPTION
**Pull request by stevejb71**

### CODE REVIEW

File: exercises/hamming/src/Hamming.idr [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Hamming
+
+import Data.Vect
+
+-- Add the correct export modifier here...
+data Nucleotide = A | C | G | T
```
Comment (yurrriq) [inline exercises/hamming/src/Hamming.idr]: Can we use a hole here too?

Comment (stevejb71) [inline exercises/hamming/src/Hamming.idr]: I would guess no, as the test requires then.

Comment (stevejb71) [inline exercises/hamming/src/Hamming.idr]: ...  requires them.

Comment (stevejb71) [inline exercises/hamming/src/Hamming.idr]: Sure, I'll try inserting a hole here instead. I think it improves the exercise.

Comment (stevejb71) [inline exercises/hamming/src/Hamming.idr]: Idris doesn't permit a hole for a constructor. 
What about an empty data type + comment:
```data Nucleotide -- insert definition here```

**Comment by yurrriq** [comment]
Ah, that makes sense. I like that approach.

**Comment by yurrriq** [comment]
LGTM, thanks. I'll try to get testing working this weekend (#3).

**Pull request merged**